src/service/impl/stop-places/index.ts

```diff
@@ -0,0 +1,107 @@
+import {IStopPlacesService} from '../../interface';
+import {journeyPlannerClient} from '../../../graphql/graphql-client';
+import {Result} from '@badrap/result';
+import {
+  GetStopPlaceConnectionsDocument,
+  GetStopPlaceConnectionsQuery,
+  GetStopPlaceConnectionsQueryVariables,
+} from './journey-gql/stop-place-connections.graphql-gen';
+import {isDefined, onlyUniquesBasedOnField} from './utils';
+import {APIError} from '../../../utils/api-error';
+import {
+  GetStopPlacesByModeDocument,
+  GetStopPlacesByModeQuery,
+  GetStopPlacesByModeQueryVariables,
+} from './journey-gql/stop-places-mode.graphql-gen';
+import {QuayFragment} from '../fragments/journey-gql/quays.graphql-gen';
+import {JourneyPatternsFragment} from '../fragments/journey-gql/journey-pattern.graphql-gen';
+
+export default (): IStopPlacesService => {
+  return {
+    async getStopPlacesByMode(query, headers) {
+      const result = await journeyPlannerClient(headers).query<
+        GetStopPlacesByModeQuery,
+        GetStopPlacesByModeQueryVariables
+      >({
+        query: GetStopPlacesByModeDocument,
+        variables: {
+          authorities: query.authorities,
+          transportModes: query.transportModes,
+        },
+      });
+
+      if (result.errors) {
+        return Result.err(new APIError(result.errors));
+      }
+
+      try {
+        const uniqueStopPlaces = result.data.lines
+          .filter((line) =>
+            query.transportSubmodes?.find(
+              (subMode) => subMode === line.transportSubmode,
+            ),
+          )
+          .flatMap((line) => line.quays)
+          .map((quay) => quay.stopPlace)
+          .filter(isDefined)
+          .filter(onlyUniquesBasedOnField('id'));
+        return Result.ok(uniqueStopPlaces);
+      } catch (error) {
+        return Result.err(new APIError(error));
+      }
+    },
+    async getStopPlaceConnections(query, headers) {
+      const result = await journeyPlannerClient(headers).query<
+        GetStopPlaceConnectionsQuery,
+        GetStopPlaceConnectionsQueryVariables
+      >({
+        query: GetStopPlaceConnectionsDocument,
+        variables: {
+          id: query.fromStopPlaceId,
+        },
+      });
+      if (result.errors) {
+        return Result.err(new APIError(result.errors));
+      }
+      if (!result.data.stopPlace?.quays) {
+        return Result.ok([]);
+      }
+      const journeyPatternsWithMatchingAuthority = result.data.stopPlace.quays
+        .flatMap((quay) => quay.journeyPatterns)
+        .filter((jp) => filterAuthorities(jp, query.authorities));
+
+      const reachableQuays = journeyPatternsWithMatchingAuthority.flatMap(
+        (jp) => getReachableQuays(jp.quays, query.fromStopPlaceId),
+      );
+
+      const reachableStopPlaces = reachableQuays.map((quay) => quay.stopPlace);
```

Comment:
I asked Entur about if we need to handle when a stop place serves different transport modes, and [their answer](https://mittatb.slack.com/archives/CHLDG8C30/p1687262055131999?thread_ts=1687160151.213509&cid=CHLDG8C30) was that it wasn't necessary.

> Nei en quay sin immediate parent stop place vil alltid være mono-modal. Om du derimot går til StopPlace.parent så kan denne ha flere modes.

However, when talking to Entur I didn't think of `transportSubmodes`. 

I think some filtering is necessary on `transportSubmodes` to ensure we don't get connections on other modes than intended. I tried looking at AtB data, and I see that for example "Sula ferjekai" serves both a `highSpeedVehiclesService` line and `localCarFerry` line, so it is something that is possible in the data model. In practice it seems that it shouldn't be a problem for AtB since other ferry stops which has a connection from Sula also serves `highSpeedVehiclesService`, but we can't be sure that this is the case also for other transport companies.

Reply:
Another interesting thing about Sula is that the ferry stops twice at the same stop, which means that it becomes a valid connection to itself.

![IMG_0EF4A01038EC-1](https://github.com/AtB-AS/atb-bff/assets/1774972/4f51d6e0-f420-4969-aa09-f7db2071bfd2)

Might need to filter out stops with `id === fromStopPlaceId`